### ISSUE TITLE
Internal compiler error on attempting to iterate over list of string literals (and nonsense error on attempting to iterate over list of strings)

### ISSUE THREAD

**Issue by haltman-at** [vyperlang/vyper#2502]
### Version Information

* vyper Version (output of `vyper --version`): 0.3.0
* OS: Linux Mint 20.2 Uma 64-bit
* Python Version (output of `python --version`): 3.8.10

### What's your issue about?

Attempting to interate over a list of string literals produces an internal compiler error.  I tried to compile the following contract:

```vyper
@external
def doThings():
    for str in ["Hello"]:
        pass
```

And got the following error message:

```
Statement node did not produce LLL

This is an unhandled internal compiler error. Please create an issue on Github to notify the developers.
https://github.com/vyperlang/vyper/issues/new?template=bug.md
```

This should either compile, or else produce a proper error message.

**Edit**:

If we change the code to

```vyper
@external
def doThings():
    text: String[64] = "Hello"
    for str in [text]:
        pass
```

then we get the same error message.  If, however, we change it to

```vyper
text: String[64]

@external
def doThings():
    self.text = "Hello"
    for str in [self.text]:
        pass
```

Then instead we get the error message:

```
'type'
```

which is still unhelpful.  This error message also should be fixed (or the code should compile).

**Comment by tserg** [comment]
I am interested in taking this up next. I should have some updates by end of the coming week.

**Comment by charles-cooper** [comment]
this should be fixed as of #2584
